components/supervisor/pkg/supervisor/notification_test.go

```diff
@@ -0,0 +1,190 @@
+package supervisor
```

Comment:
Didn't look at the tests in detail yet. Two things that come in handy for those kinds of tests are the `mock` package we usually generate for gRPC API, and writing table driven tests (e.g. [tasks_test.go](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/pkg/supervisor/tasks_test.go))

Reply:
These tests are more about the order of notifications/subscriptions that about a matrix of test data. The individual tests are pretty different in structure as we have to handle different blocking scenarios. I think it's easier to leave them as they are.